OpenRA.Mods.Common/Traits/Infantry/TakeCover.cs

```diff
@@ -66,7 +87,12 @@ public override void Tick(Actor self)
 
 		public int GetDamageModifier(Actor attacker, DamageWarhead warhead)
 		{
-			return IsProne && warhead != null ? warhead.ProneModifier : 100;
+			if (!IsProne)
+				return 100;
+
+			var modifiedTypes = warhead.DamageTypes.Where(x => info.DamageModifiers.ContainsKey(x));
+			var modifier = modifiedTypes.Aggregate(100, (current, modifiedType) => current*info.DamageModifiers[modifiedType]);
```

Comment:
This needs to be using `return Util.ApplyPercentageModifiers(100, modifierPercentages)`